teuthology/orchestra/daemon/systemd.py

```diff
@@ -63,9 +63,13 @@ def check_status(self):
         def parse_line(line):
             key, value = line.strip().split('=', 1)
             return {key.strip(): value.strip()}
+
         show_dict = dict()
-        for line in output.split('\n'):
+
+        # strip out newline character 
+        for line in output.strip().split('\n'):
```

Comment:
@sunilkumarn417 what was the symptom of the issue addressed by this change ?

Reply:
@tchaikov  when systemctl command output splitted into lines, a empty string was part of list due to newline character at end of the string which was causing issue in parsing.
```
> /home/sunnagar/fork/teuthology/teuthology/orchestra/daemon/systemd.py(63)check_status()
-> def parse_line(line):
(Pdb) output
'StateDirectoryMode=0755\nLoadState=loaded\nActiveState=active\nSubState=running\nUnitFileState=enabled\nStateChangeTimestamp=Thu 2020-06-18 09:20:05 UTC\nStateChangeTimestampMonotonic=913770846\n'
(Pdb) n
...
...
> /home/sunnagar/fork/teuthology/teuthology/orchestra/daemon/systemd.py(63)parse_line()
-> def parse_line(line):
(Pdb) n
> /home/sunnagar/fork/teuthology/teuthology/orchestra/daemon/systemd.py(64)parse_line()
-> key, value = line.strip().split('=', 1)
(Pdb) line
''
(Pdb) line.strip().split('=', 1)
['']
...
...
  File "/home/sunnagar/fork/teuthology/teuthology/orchestra/daemon/systemd.py", line 145, in restart
    self.check_status()
  File "/home/sunnagar/fork/teuthology/teuthology/orchestra/daemon/systemd.py", line 68, in check_status
    show_dict.update(parse_line(line))
  File "/home/sunnagar/fork/teuthology/teuthology/orchestra/daemon/systemd.py", line 64, in parse_line
    key, value = line.strip().split('=', 1)
ValueError: not enough values to unpack (expected 2, got 1) 
```

Reply:
http://pastebin.test.redhat.com/876548

Reply:
@sunilkumarn417 instead of this, I would suggest to update parse_line function to skip empty string and strings commented with '#' character.

Reply:
@kshtsk added changes in the loop and eliminating empty, commented line.
```
>>> for line in s.split("\n"):
...    if not line or line.startswith("#"):
...        print("Not considered for parsing\t\t: '%s'" % line)
...        continue
...    print("Considered for parsing\t\t: '%s'" % line)
... 
Considered for parsing		: 'StateDirectoryMode=0755'
Considered for parsing		: 'LoadState=loaded'
Considered for parsing		: 'ActiveState=active'
Considered for parsing		: 'SubState=running'
Considered for parsing		: 'UnitFileState=enabled'
Considered for parsing		: 'StateChangeTimestamp=Thu 2020-06-18 09:20:05 UTC'
Considered for parsing		: 'StateChangeTimestampMonotonic=913770846'
Not considered for parsing		: '#test'
Not considered for parsing		: ''
```

Reply:
looks good

Reply:
interesting, stripping is not bad idea though, like:
```
for line in (_.strip() for _ in s.split("\n")):
. . .
```

Reply:
oh... this is parsing the status output for ceph specific services, sorry,  I didn't realize it, so maybe add corresponding unit test is better.